packages/firestore/lib/index.d.ts

```diff
@@ -237,7 +237,9 @@ export namespace FirebaseFirestoreTypes {
      *
      * @param collectionPath A slash-separated path to a collection.
      */
-    collection(collectionPath: string): CollectionReference;
+    collection<T extends DocumentData = DocumentData>(
+      collectionPath: string,
+    ): CollectionReference<T>;
```

Comment:
Normally we try to match the types from firebase-js-sdk, I am consulting those and I don't see the generics passed in but I welcome a pointer to a better set of reference lines, I can definitely be missing something here:

https://github.com/firebase/firebase-js-sdk/blob/e650f649854f3c39737fe4bade43f9eedc3e611f/packages/firestore-types/index.d.ts#L252

They have generics in a bunch of other places, I wonder why not here? Until their types move I am cautious on moving ours, as I consider their types "upstream" with ours derivative.

Have you brought this up with firebase-js-sdk?

Reply:
No. I didn't know that typings here were from firebase-js-sdk. 
I think that I should make a pull request on firebase-js-sdk repository first.